src/main/java/com/onarandombox/multiverseinventories/util/CommentedYamlConfiguration.java

```diff
@@ -66,44 +69,36 @@ public boolean save() {
         if (!doComments) {
             return saved;
         }
+
         // if there's comments to add and it saved fine, we need to add comments
         if (!comments.isEmpty() && saved) {
-            // String array of each line in the config file
-            String[] yamlContents =
-                    this.convertFileToString(file).split("[" + System.getProperty("line.separator") + "]");
-            // This will hold the entire newly formatted config
+            // convert config file to String
+            String stringConfig = this.convertFileToString(file);
+
+            // detect which kind of line endings it uses
+            String lineEnding;
+            if (stringConfig.contains("\r\n")) lineEnding = "\r\n";
+            else lineEnding = "\n";
```

Comment:
I added this to make it so that if a user moves their config files from a system using CRLF to a system using LF (or vice versa), they'd be converted. But now I'm wondering if the server does this automatically. What do you think?

Reply:
> But now I'm wondering if the server does this automatically. What do you think?

If by "converted" you mean "the server reads them with error" = true. If you mean "files are converted to the type of the system" = false. At least in the DOS->Linux direction (ie., on Linux servers), files are not converted automatically. I convert every DOS-produced file with `dos2unix`. But I do that only for my own convenience and sanity; conversion is not required for the files to be read correctly.

Reply:
I see... if it's not done automatically, maybe we shouldn't be doing either.

Reply:
Looking at the source code of the [YamlConfiguration](https://hub.spigotmc.org/stash/projects/SPIGOT/repos/bukkit/browse/src/main/java/org/bukkit/configuration/file/YamlConfiguration.java#119) class, it seems Spigot has chosen to make all config files use the LF line ending, so we will do the the same.